Products/Zuul/facades/metricfacade.py

```diff
@@ -373,3 +453,70 @@ def getMetricsByDevice(self, device, metrics, start=None,
             if status['status'] == u'ERROR' and 'not such name' not in status['message']:
                 log.error(status['message'])
         return content['series']
+
+    def _getDataPoint(self, devices, metric):
+        for subject in devices:
+            dp = next(
+                (
+                    d
+                    for d in subject._getRRDDataPointsGen()
+                    if metric in d.name()
+                ),
+                None
+            )
+            if dp is not None:
+                return dp
+
+        return None
+
+    def getMetricsForDevices(self, devices, metrics, start=None,
+                             end=None, format="%.2lf", cf="avg",
+                             downsample=None, timeout=10, isRate=False):
+        """
+        Gets a set of metrics for every component under a each device.
+        """
+        # Only EXACT resultsets are supported yet.
+        returnSet = 'EXACT'
+
+        start, end = self._defaultStartAndEndTime(start, end, returnSet)
+        if isinstance(metrics, basestring):
+            metrics = [metrics]
+
+        metricnames = {}
+        metricRequests = []
+        for device in devices:
+            for metric in metrics:
+                dp = self._getDataPoint(
+                    itertools.chain((device, ), device.getDeviceComponents()),
+                    metric)
+                if dp is not None:
+                    metricnames[dp.name()] = metric
+                    metricRequests.append(
+                        self._buildWildCardMetrics(device, dp.name(), cf, isRate,
+                                                   format))
+
+        request = self._buildRequest(devices, metricRequests, start, end, returnSet, downsample)
+        request['queries'] = request['metrics']
+        del request['metrics']
+        content = self._metrics_connection.request(WILDCARD_URL_PATH,
+                                                   request,
+                                                   timeout=timeout)
+        if content is None:
+            return {}
+
+        # check for bad status and log what happened
+        for status in content['statuses']:
+            if status['status'] == u'ERROR' and 'No such name' not in status['message']:
```

Comment:
Is the value for `status['message']` also in unicode?  If it is, I think this second clause in the `if` statement is always false.